src/Data/Rational/Properties.agda

```diff
@@ -55,6 +55,16 @@ private
   _≢0 : ℕ → Set
   n ≢0 = False (n ℕ.≟ 0)
 
+------------------------------------------------------------------------
+-- Properties of Positive/Negative/NonPositive/NonNegative predicates
+------------------------------------------------------------------------
+
+positive⇒nonNegative : ∀ {q} → Positive q → NonNegative q
```

Comment:
So for rationals, the underlying number cannot be inferred just from `Positive/Negative etc`. This means that you need to make the `q` argument explicit rather than implicit in many places. I realise we haven't been great about enforcing that so far in this file, but better to start now!